src/harness/testcases/WINNF_FT_S_PAT_testcase.py

```diff
@@ -0,0 +1,124 @@
+#    Copyright 2017 SAS Project Authors. All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+from datetime import datetime
+import json
+import logging
+import os
+from os import listdir
+from os.path import isfile, join
+import sas
+import sas_testcase
+
+from util import winnforum_testcase, configurable_testcase, writeConfig, loadConfig, QueryPropagationAntennaModel
+  
+def cbsddata(device):
+    installationParam = device['installationParam']
+    if 'antennaAzimuth' not in installationParam:
+        installationParam['antennaAzimuth'] = None
+        
+    cbsd = {'latitude': installationParam['latitude'],
+             'longitude': installationParam['longitude'],
+              'height': installationParam['height'],
+              'heightType': installationParam['heightType'],
+              'indoorDeployment': installationParam['indoorDeployment'],
+              'antennaAzimuth': installationParam['antennaAzimuth'],
+              'antennaGain': installationParam['antennaGain'],
+              'antennaBeamwidth': installationParam['antennaBeamwidth'],
+              'cbsdCategory': device['cbsdCategory']}
+    return cbsd
+
+def fssdata(fss_record, rxAntennaGainRequired=None):
+    deploymentParam = fss_record['deploymentParam'][0]
+    installationParam = deploymentParam['installationParam']
+    if 'antennaAzimuth' not in installationParam:
+        installationParam['antennaAzimuth'] = ''
```

Comment:
Is this valid?

Reply:
I dont think it is valid. This will crash the antenna code, which expect a valid numerical value.
If all the required parameters are not passed correctly then you cannot do your job and you should raise an exception here (but probably only when  rxAntennaGainRequired=False, because in that case it is not required to apply the FSS antenna).
In general you should do all sanity checks at this level for a robust implementation.